LayoutTests/ipc/coreipc.js

```diff
@@ -491,32 +491,56 @@ export class ArgumentSerializer {
     }
 
     static serializeKeyValuePair(innerType, argument) {
-        const splitTemplateType = ArgumentSerializer.splitTemplateType(innerType);
-        if(argument['key'] === undefined)
-            throw new SerializationError('argument of key value type is missing the key attribute');
-        if(argument['value'] === undefined)
-            throw new SerializationError('argument of key value type is missing the key attribute');
-        const result = [];
-        let argumentDefinition = {type: splitTemplateType[0], name: "key"};
-        result.push(ArgumentSerializer.serializeArgument(argumentDefinition, argument['key']));
-        argumentDefinition = {type: splitTemplateType[1], name: "value"};
-        result.push(ArgumentSerializer.serializeArgument(argumentDefinition, argument['value']));
-        return result;
+        // we handle both [key, value] and {key, value} formats
+
+        if(Array.isArray(argument) && argument.length == 2) {
+            return ArgumentSerializer.serializePair(innerType, argument);
+        }
+
+        if(argument['key'] !== undefined && argument['value'] !== undefined) {
```

Comment:
I think we normally early return on error, like:
```
if (argument['key'] === undefined || argument['value'] === undefined)
    throw new SerializationError(...)

const splitTemplateType = ArgumentSerializer.splitTemplateType(innerType);
...
```

(And I think there should be space between `if` and `(`, but it seems all conditions are missing that)